druid/src/widget/flex.rs

```diff
@@ -130,6 +171,23 @@ impl<T> Flex<T> {
         self
     }
 
+    /// Builder-style method for specifying the childrens' [`Justification`].
+    ///
+    /// [`Justification`]: enum.Justification.html
+    pub fn justification(mut self, justification: Justification) -> Self {
+        self.justification = justification;
+        self
+    }
+
+    /// Builder-style method for setting whether the container must expand
```

Comment:
For some reason I'm having a hard time understanding this. I guess because I assumed this is what a Flex widget was doing already?

Reply:
yea, I can understand the confusion.

Let's say a flex row is passed a width constraint `(0., 400.)`. None if its children are flex. Is it still obliged to have a final width of `400.0`, or can it be smaller? This controls that behaviour.

I'll think about how to make this clearer.

Reply:
Gotcha.

If any children are flex this Flex container will expand to fill all available space on its main axis. But if no children are flex, `fill_main_axis` determines whether or not Flex should shrink to fit, or still fill its main axis (TKTK something about what it does with the extra space?)